graphene/src/layers/text/to_kurbo.rs

```diff
@@ -0,0 +1,168 @@
+				buzz_face.outline_glyph(GlyphId(glyph_info.glyph_id as u16), &mut builder);
+				builder.pos += Vec2::new(glyph_position.x_advance as f64, glyph_position.y_advance as f64) * builder.scale;
+			}
+
+			buffer = glyph_buffer.clear();
+		}
+		builder.pos = Point::new(0., builder.pos.y + line_height);
+	}
+	builder.path
+}
+
+pub fn bounding_box(str: &str, buzz_face: rustybuzz::Face, font_size: f64, line_width: Option<f64>) -> DVec2 {
+	let (scale, line_height, mut buffer) = font_properties(&buzz_face, font_size);
+
+	let mut pos = DVec2::ZERO;
+	let mut bounds = DVec2::ZERO;
+
+	for line in str.split('\n') {
+		let length = line.split(' ').count();
+		for (index, word) in line.split(' ').enumerate() {
+			push_str(&mut buffer, word, index != length - 1);
+
+			let glyph_buffer = rustybuzz::shape(&buzz_face, &[], buffer);
+
+			if wrap_word(line_width, &glyph_buffer, scale, pos.x) {
+				pos = DVec2::new(0., pos.y + line_height);
+			}
+
+			for glyph_position in glyph_buffer.glyph_positions() {
+				if let Some(line_width) = line_width {
+					if pos.x + (glyph_position.x_advance as f64 * scale) >= line_width {
+						pos = DVec2::new(0., pos.y + line_height);
+					}
+				}
+				pos += DVec2::new(glyph_position.x_advance as f64, glyph_position.y_advance as f64) * scale;
+			}
+			bounds = bounds.max(pos + DVec2::new(0., line_height));
+
+			buffer = glyph_buffer.clear();
+		}
+		pos = DVec2::new(0., pos.y + line_height);
+	}
+
+	bounds
+}
+
+#[test]
+fn test() {
+	use std::fs::File;
+	use std::io::Write;
+
+	let buzz_face = rustybuzz::Face::from_slice(include_bytes!("SourceSansPro/SourceSansPro-Regular.ttf"), 0).unwrap();
+
+	let text = r#"The quick brown
+fox jumped over the lazy cat.
+In publishing and graphic design, Lorem ipsum is a placeholder text commonly used to demonstrate the visual form of a document or a typeface without relying on meaningful content. Lorem ipsum may be used as a placeholder before the final copy is available. It is also used to temporarily replace text in a process called greeking, which allows designers to consider the form of a webpage or publication, without the meaning of the text influencing the design.
+
+Lorem ipsum is typically a corrupted version of De finibus bonorum et malorum, a 1st-century BC text by the Roman statesman and philosopher Cicero, with words altered, added, and removed to make it nonsensical and improper Latin.
+
+Test for really long word: aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa"#;
+	let svg = to_kurbo(text, buzz_face, 20., Some(400.)).to_svg();
+
+	let mut file = File::create("src/layers/text/SourceSansPro/font_text.svg").unwrap();
+	write!(
+		&mut file,
+		r#"<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink"><path d="{}" /></svg>"#,
+		svg
+	)
+	.unwrap();
```

Comment:
Could we create a temp file here instead? using a relative path might also potentially break I think?

Reply:
Actually, I'm a bit confused, is this a test that is supposed to be checked manually? If so it should probably not run as part of the regular test suite

Reply:
This was just me messing around with harfbuzz. I will remove it.